File: InMotionGIT.Common - Copy/Base/Action.vb
Imports System.ComponentModel
Imports System.Xml.Serialization

Namespace Base

    <Serializable()>
    <XmlType(Namespace:="urn:InMotionGIT.UI.Model")>
    <XmlRoot(Namespace:="urn:InMotionGIT.UI.Model")>
    Public MustInherit Class Action

#Region "Public Properties"

        <XmlAttribute()>
        <BrowsableAttribute(False)>
        Public Property ControlName As String

#End Region

        Public MustOverride Function Description() As String

#Region "Functions"

        Public MustOverride Function Clone() As Action

#End Region

    End Class

End Namespace

File: Common/InMotionGIT.Common/Base/Action.cs
using System;
using System.ComponentModel;
using System.Xml.Serialization;

namespace InMotionGIT.Common.Base
{

    [Serializable()]
    [XmlType(Namespace = "urn:InMotionGIT.UI.Model")]
    [XmlRoot(Namespace = "urn:InMotionGIT.UI.Model")]
    public abstract class Action
    {

        #region Public Properties

        [XmlAttribute()]
        [Browsable(false)]
        public string ControlName { get; set; }

        #endregion

        public abstract string Description();

        #region Functions

        public abstract Action Clone();

        #endregion

    }

}